### DISCUSSION TITLE
What are the implications setting enable_storage_cache to true

### DISCUSSION THREAD

**Discussion by porscheme** [vesoft-inc/nebula#4691] [Q&A]
Can someone please explain what are the implications setting enable_storage_cache to true, does setting enable_storage_cache to true cache entire storage?

Also, what happens if enable_vertex_pool and enable_empty_key_pool was set true?

The [Official docs](https://docs.nebula-graph.io/3.2.1/5.configurations-and-logs/1.configurations/4.storage-config/#storage_cache_configurations) are not clear.

Below are the setting in our conf file
```
############## storage cache ##############
############## storage cache ##############
# Whether to enable storage cache
--enable_storage_cache=false

# Total capacity reserved for storage in memory cache in MB
--storage_cache_capacity=0

# Number of buckets in base 2 logarithm. E.g., in case of 20, the total number of buckets will be 2^20.
# A good estimate can be ceil(log2(cache_entries * 1.6)). The maximum allowed is 32.
--storage_cache_buckets_power=20

# Number of locks in base 2 logarithm. E.g., in case of 10, the total number of locks will be 2^10.
# A good estimate can be max(1, buckets_power - 10). The maximum allowed is 32.
--storage_cache_locks_power=10

# Whether to add vertex pool in cache. Only valid when storage cache is enabled.
--enable_vertex_pool=false

# Vertex pool size in MB
--vertex_pool_capacity=50

# TTL in seconds for vertex items in the cache
--vertex_item_ttl=300

# Whether to add empty key pool in cache. Only valid when storage cache is enabled.
--enable_empty_key_pool=false

# Empty key pool size in MB
--empty_key_pool_capacity=50

# TTL in seconds for empty key items in the cache
--empty_key_item_ttl=300
```

**Comment by wenhaocs** [comment]
Hi, `enable_storage_cache ` only caches two things, vertex and negative keys, corresponding to `vertex_pool` and `empty_key_pool` respectively. `vertex_pool` caches the vertex you queried which exists in the storage. It's useful when you need to query the same keys in a short period. `empty_key_pool` caches the vertex you queried but does not exist in the storage. It's can be useful in many cases, e.g., you want to get the property of a vertex without explicitly specifying which tag your are looking for.
